Sources/Workspace/DefaultPluginScriptRunner.swift

```
@@ -179,35 +182,80 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner {
+#if os(Windows)
+            let execSuffix = ".exe"
+#else
+            let execSuffix = ""
+#endif
+            let executableFile = cacheDir.appending(component: execName + execSuffix)
             command += ["-o", executableFile.pathString]
         
             // Create the cache directory in which we'll be placing the compiled executable if needed.
             try FileManager.default.createDirectory(at: cacheDir.asURL, withIntermediateDirectories: true, attributes: nil)
         
-            // Compile the plugin script asynchronously.
-            Process.popen(arguments: command, environment: toolchain.swiftCompilerEnvironment, queue: callbackQueue) {
-                // We are now on our caller's requested callback queue, so we just call the completion handler directly.
-                dispatchPrecondition(condition: .onQueue(callbackQueue))
-                completion($0.tryMap {
-                    // Emit the compiler output as observable info.
-                    let compilerOutput = ((try? $0.utf8Output()) ?? "") + ((try? $0.utf8stderrOutput()) ?? "")
-                    observabilityScope.emit(info: compilerOutput)
-
-                    // We return a PluginCompilationResult for both the successful and unsuccessful cases (to convey diagnostics, etc).
-                    let result = PluginCompilationResult(compilerResult: $0, diagnosticsFile: diagnosticsFile, compiledExecutable: executableFile)
-                    guard $0.exitStatus == .terminated(code: 0) else {
-                        throw DefaultPluginScriptRunnerError.compilationFailed(result)
-                    }
-                    return result
-                })
+            // Hash the command line and the contents of the source files to decide whether we need to recompile the plugin executable. We have a bounded set of input conditions, and can control what we hash.
+            let compilerInputsHash: String? = try? {
```

is try? hiding error we should report?

---

Agreed.  Fixed.